src/mainwindow.cpp

```diff
@@ -938,19 +940,25 @@ QString MainWindow::getLastDir() const
 }
 
 void MainWindow::setNameFilters()
-{
-    nameFilters << "*.png" << "*.pbm"
-    << "*.jpg" << "*.jpeg"
-    << "*.gif" << "*.tiff"
-    << "*.xbm" << "*.JPG"
-    << "*.bmp" << "*.ppm"
-    << "*.pgm" << "*.mng"
-    << "*.xpm"
+{   
+    /*
+     * Use known formats instead of hardcoded ones.
+     */
+    
+    foreach(QByteArray format, QImageReader::supportedImageFormats()) {
+        nameFilters << QString("*." + format);
+    }
+    
     #ifdef WEBP_SUPPORT
-    << "*.webp"
+    nameFilters << "*.webp"
```

Comment:
You've missed a ";". The compiling error comes out only if you have enabled the feature.